synapse/util/caches/lrucache.py

```diff
@@ -788,26 +788,21 @@ def __len__(self) -> int:
     def __contains__(self, key: KT) -> bool:
         return self.contains(key)
 
-    def set_cache_factor(self, factor: float) -> bool:
```

Comment:
The return value is not used and doesn't match what `add_resizable_cache` expects.

Reply:
Could also update this to match: https://github.com/matrix-org/synapse/blob/d8cc86eff484b6f570f55a5badb337080c6e4dcd/synapse/util/caches/__init__.py#L183 (though you might consider out-of-scope)

Reply:
Alternatively: we could change `add_resizable_cache` to expect a callable that returns Any, as a way to say "I don't care what you return". (Not sure if that's the done thing?)

Reply:
> Alternatively: we could change `add_resizable_cache` to expect a callable that returns Any, as a way to say "I don't care what you return". (Not sure if that's the done thing?)

I debated this, but there are no other callers to `set_cache_factor` so I think it is really meant to be used only with `add_resizable_cache`?

Reply:
Maybe. I guess that might change if we ever did some auto cache resizing stuff? But let's leave this as it is in the PR right now. :+1: